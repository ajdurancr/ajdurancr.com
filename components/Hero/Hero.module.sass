@import '../../styles/_variables'
  
.container
  display: flex
  height: 100vh
  margin-top: -$menuHeightMobile
  position: relative
  width: 100%

  .imageContainer
    background: transparent center no-repeat
    background-size: cover
    height: 100%
    position: absolute
    width: 100%
