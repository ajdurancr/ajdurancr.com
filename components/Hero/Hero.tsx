import React from 'react'
import classnames from 'classnames'
import get from 'lodash.get'

import styles from './Hero.module.sass'
import { CfMedia } from '../../types/contentful'

type HeroPros = {
  className?: string
  image?: CfMedia
};

const Hero: React.FC<HeroPros> = ({ className = '', image }) => (
  <div className={classnames(styles.container, className)}>
    <div
      className={classnames(styles.imageContainer)}
      style={{ backgroundImage: `url(${get(image, 'fields.file.url')})` }}
    />
  </div>
)

export { Hero }
