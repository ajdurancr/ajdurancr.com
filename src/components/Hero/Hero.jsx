import React from 'react'
import propTypes from 'prop-types'
import classnames from 'classnames'
import get from 'lodash.get'

import styles from './Hero.module.sass'

const Hero = ({ className, image }) => (
  <div className={classnames(styles.container, className)}>
    <div
      className={classnames(styles.imageContainer)}
      style={{ backgroundImage: `url(${get(image, 'fields.file.url')})` }}
    />
  </div>
)

Hero.defaultProps = {
  className: '',
}

Hero.propTypes = {
  className: propTypes.string,
  image: propTypes.object.isRequired,
}

export default Hero
