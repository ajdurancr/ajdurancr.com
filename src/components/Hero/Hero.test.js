import React from 'react'
import { shallow } from 'enzyme'

import Hero from './Hero'
import image from '../../mocks/contentful/media'

describe('Hero', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Hero className="testClassName" image={{ fields: image }} />)

    expect(wrapper).toMatchSnapshot()
  })
})
