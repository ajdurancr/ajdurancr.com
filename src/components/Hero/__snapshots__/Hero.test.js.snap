// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div
  className="container testClassName"
>
  <div
    className="imageContainer"
    style={
      Object {
        "backgroundImage": "url(//images.ctfassets.net/52uxn1pv2fou/5Yt9H0zn6ewSMnuvzh7V6P/3d6a253c1dce0250c70ef3a52c169872/github.jpg)",
      }
    }
  />
</div>
`;
